src/sensor.cpp

```diff
@@ -441,10 +429,23 @@ void log_callback_end( uint32_t fps,
 
                     if (fh.frame)
                     {
-                        assert( expected_size == sizeof(byte) * fr->data.size() ||
-                                expected_size == sizeof(byte) * fr->data.size() + 68); // added for D457 - need to understand why this happens (68 is size of md)
-
-                        memcpy((void*)fh->get_frame_data(), fr->data.data(), sizeof(byte) * fr->data.size());
+                        // method should be limited to use of MIPI - not for USB
+                        // the aim is to grab the data from a bigger buffer, which is aligned to 64 bytes,
+                        // when the resolution's width is not aligned to 64
+                        if (width % 64 != 0 && f.frame_size > expected_size)
+                        {
+                            byte* pix = (byte*)f.pixels;
+                            std::vector<byte> pixels;
+                            pixels = align_width_to_64(width, height, bpp, pix);
+                            fr->data = pixels;
```

Comment:
I think the second line can be skipped and save one memcpy

Reply:
Done